app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -147,7 +151,7 @@ open class HomeActivity : AppCompatActivity() {
         // Toggle off the open_link_in_private_tab pref if we are no longer set as the default browser
         // We do this on a separate thread to alleviate performance issues
         val weakReferenceContext = WeakReference(this)
-        lifecycleScope.launch {
+        scope.launch {
```

Comment:
To change which thread is used, we need to change the dispatcher rather than the coroutine scope. The scope defines when the work will be cancelled (when the activity is destroyed in the case of `lifecycleScope`), and the dispatcher defines the thread. `Dispatchers.Main` will keep running code on the main thread.

```suggestion
        lifecycleScope.launch(Dispatchers.IO) {
```